web/src/main/java/com/epam/rft/atsy/web/controllers/rest/StateHistoryController.java

```diff
@@ -54,37 +52,30 @@ public ResponseEntity saveOrUpdate(@RequestParam Long applicationId,
     if (bindingResult.hasErrors()) {
       return fieldErrorResponseComposer.composeResponse(bindingResult);
     } else {
-      try {
-        stateHistoryDTO = StateHistoryDTO.builder()
-            .id(stateHistoryViewRepresentation.getId())
-            .candidateId(stateHistoryViewRepresentation.getCandidateId())
-            .description(stateHistoryViewRepresentation.getDescription())
-            .result(stateHistoryViewRepresentation.getResult())
-            .offeredMoney(stateHistoryViewRepresentation.getOfferedMoney())
-            .claim(stateHistoryViewRepresentation.getClaim())
-            .feedbackDate(stateHistoryViewRepresentation.getFeedbackDate() != null
-                && !stateHistoryViewRepresentation.getFeedbackDate().isEmpty() ? DATE_FORMAT
-                .parse(stateHistoryViewRepresentation.getFeedbackDate()) : null)
-            .dayOfStart(stateHistoryViewRepresentation.getDayOfStart())
-            .dateOfEnter(stateHistoryViewRepresentation.getDateOfEnter())
-            .creationDate(null)
-            .stateDTO(StateDTO.builder().id(stateHistoryViewRepresentation.getStateId())
-                .name(stateHistoryViewRepresentation.getStateName()).build())
-            .recommendation(stateHistoryViewRepresentation.getRecommendation() != null ?
-                stateHistoryViewRepresentation.getRecommendation() == 1 : null)
-            .reviewerName(stateHistoryViewRepresentation.getReviewerName())
-            .recommendedPositionLevel(stateHistoryViewRepresentation.getRecommendedPositionLevel())
-            .build();
-      } catch (ParseException e) {
-        RestResponse restResponse = new RestResponse(e.getMessage());
-
-        return new ResponseEntity<>(restResponse, HttpStatus.BAD_REQUEST);
-      }
+      stateHistoryDTO = StateHistoryDTO.builder()
+          .id(stateHistoryViewRepresentation.getId())
+          .candidateId(stateHistoryViewRepresentation.getCandidateId())
+          .description(stateHistoryViewRepresentation.getDescription())
+          .result(stateHistoryViewRepresentation.getResult())
+          .offeredMoney(stateHistoryViewRepresentation.getOfferedMoney())
+          .claim(stateHistoryViewRepresentation.getClaim())
+          .feedbackDate(stateHistoryViewRepresentation.getFeedbackDate() != null
+              ? stateHistoryViewRepresentation.getFeedbackDate() : null)
```

Comment:
I do not really understand why is this expression is necessary. If you simply use stateHistoryViewRepresentation.getFeedbackDate(), then null value will be set if feedback date is null.